ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -115,8 +116,20 @@ object SqlLedgerReaderWriter {
           validator = validator,
           postCommit = dispatcher.signalNewHead,
         )
+        committerExecutionContext <- ResourceOwner
+          .forExecutorService(() =>
+            ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor()))
```

Comment:
If I understand correctly, this will be also used by the ledger state in direct mode. I assume it worked correctly for you with DAML-on-SQL but there were deadlocks when I attempted to do so in DAML-on-Memory so I'd test this carefully. Also, any assessment on the performance implications?

Reply:
Hmmm, not sure how to test it past running the conformance tests (which work fine).

Performance isn't impacted at all.

My understanding is that because [we only allocate a single executor for writing](https://github.com/digital-asset/daml/pull/7063/files#diff-1454db978c0fa4672d935447db837b38R77) anyway, this simply ensures the committer _also_ has a dedicated thread.